src/dbt/common.py

```diff
@@ -0,0 +1,126 @@
+"""Shared methods between both sync scripts."""
+
+from datetime import datetime, timezone
+from typing import List, Tuple
+from web3 import Web3
+from src.logger import set_log
+
+log = set_log(__name__)
+
+
+def node_suffix(network: str) -> str:
+    """
+    Converts network internal name to name used for nodes and dune tables
+    """
+    if network == "mainnet":
+        return "ETHEREUM"
+    if network == "xdai":
+        return "GNOSIS"
+    if network == "arbitrum-one":
+        return "ARBITRUM"
+    return ""
+
+
+def find_block_with_timestamp(node: Web3, time_stamp: float) -> int:
+    """
+    This implements binary search and returns the smallest block number
+    whose timestamp is at least as large as the time_stamp argument passed in the function
+    """
+    end_block_number = int(node.eth.get_block("finalized")["number"])
+    start_block_number = 1
+    close_in_seconds = 30
+
+    while True:
+        mid_block_number = (start_block_number + end_block_number) // 2
+        block = node.eth.get_block(mid_block_number)
+        block_time = block["timestamp"]
+        difference_in_seconds = int((time_stamp - block_time))
+
+        if abs(difference_in_seconds) < close_in_seconds:
+            break
+
+        if difference_in_seconds < 0:
+            end_block_number = mid_block_number - 1
+        else:
+            start_block_number = mid_block_number + 1
+
+    ## we now brute-force to ensure we have found the right block
+    for b in range(mid_block_number - 200, mid_block_number + 200):
+        block = node.eth.get_block(b)
+        block_time_stamp = block["timestamp"]
+        if block_time_stamp >= time_stamp:
+            return int(block["number"])
+    # fallback in case correct block number hasn't been found
+    # in that case, we will include some more blocks than necessary
+    return mid_block_number + 200
+
+
+def compute_block_and_month_range(  # pylint: disable=too-many-locals
+    node: Web3, recompute_previous_month: bool
+) -> Tuple[List[Tuple[int, int]], List[str], List[bool]]:
+    """
+    This determines the block range and the relevant months
+    for which we will compute and upload data on Dune.
+    """
```

Comment:
This function might benefit from documenting the return value in a bit more detail.